iam/sc-enduser-iam.yaml

```diff
@@ -30,12 +30,10 @@ Resources:
         Statement:
           - Effect: Allow
             Principal:
-              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
-              # Synapse OIDC identity provider
-              Federated: !Sub 'arn:aws:iam::${AWS::AccountId}:oidc-provider/repo-prod.prod.sagebase.org/auth/v1'
+              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:role/synapse-login-scipool*-InstanceRole-*'
```

Comment:
if this works then i'm fine with it.  Just wondering if this is a use case for using the `Condition` property instead of `*`?  https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-idp_oidc.html

Reply:
Are you suggesting using "StringLike" rather than *?  I'm happy to do so if that's a more standard way to filter.  

(Note that the page you linked is for connecting an OIDC provider, which we are no longer doing.)

Reply:
i guess i like the way that works.

Reply:
I just realized I should try this "manually" in scipool dev before we merge and deploy the PR.  The result was:
```
An error occurred: Invalid principal in policy: "AWS":"arn:aws:iam::465877038949:role/synapse-login-scipool*-InstanceRole-*"
```
`"arn:aws:iam::465877038949:role/*"` didn't work either:
```
An error occurred: Invalid principal in policy: "AWS":"arn:aws:iam::465877038949:role/*"
```
So perhaps wildcards are not allowed.

Reply:
I will close this PR and open a new one once I verify it works via manual deployment.